cmake/Platform/Core/BoardPropertiesReader.cmake

```diff
@@ -28,6 +28,34 @@ function(_GET_BOARD_ID BOARD_NAME BOARD_CPU TARGET_NAME OUTPUT_VAR)
     endif()
 endfunction()
 
+#=============================================================================#
+# _recursively_replace_properties
+# [PRIVATE/INTERNAL]
+#
+# _recursively_replace_properties(BOARD_ID PROPERTY_VALUE_VAR)
+#        BOARD_ID - return value from function "_get_board_id (BOARD_NAME, BOARD_CPU)". It contains BOARD_NAME and BOARD_CPU
+#        PROPERTY_VALUE_VAR - the value of a property that may contain
+#           references to other properties.
+#
+# Recursively replaces references to other properties.
+#
+#=============================================================================#
+function(_recursively_replace_properties BOARD_ID PROPERTY_VALUE_VAR)
+   while("${${PROPERTY_VALUE_VAR}}" MATCHES "[^\$]{([^}]*)}")
```

Comment:
This regular expression is so, so, complicated. Could you at least comment beside what it should match?

Reply:
Apologies. This regular expressions looks for arduino property references that are `{property_name}` the `[^\$]` part is just there to ensure that something like `${foo}` is not matched as it could be a shell variable or a cmake variable or whatever, but not a Arduino property.

Reply:
Could you please add corresponding comment in the code? After all this conversation is going to disappear some time...